### PR TITLE
Quick Fix on #126854, deepcopy `lr` and other possible `base_parameters`

### LINKED ISSUE #126854: LambdaLR has incorrect multiplicative behavior when using torch.tensor LR
### 🐛 Describe the bug

```
import torch.nn as nn
import torch.optim as optim
import torch.optim.lr_scheduler as lr_scheduler
import torch

model = nn.Linear(10, 1)

optimizer = optim.Adam(model.parameters(), lr=torch.ones(1))

lambda1 = lambda epoch: 0.5
lr_scheduler = lr_scheduler.LambdaLR(optimizer, lambda1)

epochs = 10
for epoch in range(epochs):
    optimizer.step()
    lr_scheduler.step()
    print(optimizer.state_dict()["param_groups"][0]["lr"])
```

Output:
```
tensor([0.2500])
tensor([0.1250])
tensor([0.0625])
tensor([0.0312])
tensor([0.0156])
tensor([0.0078])
tensor([0.0039])
tensor([0.0020])
tensor([0.0010])
tensor([0.0005])
```
They should all be 0.5.

### Versions

```
PyTorch version: 2.4.0a0+ad67553
```
Does not appear on 2.3 or when using float LR.

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by GdoongMathew**
* Apply `deepcopy` to every base parameters (`initial_lr`, `max_lr`) when instantiating `LRScheduler`.

Fixes #126854

### CODE REVIEW

**Comment by GdoongMathew** [comment]
not sure if this is going to cause any problems, but when I run `lintrunner test/optim/test_lrscheduler.py torch/optim/lr_scheduler.py -a`, it shows the following warnings on both files.

```
>>> Lint for test/optim/test_lrscheduler.py:

  Warning (FLAKE8) EXE002
    The file is executable but no shebang is present.
    See

    >>>    1  |# Owner(s): ["module: optimizer", "module: LrScheduler" ]
           2  |import copy
           3  |import math
           4  |import pickle



>>> Lint for torch/optim/lr_scheduler.py:

  Warning (FLAKE8) EXE002
    The file is executable but no shebang is present.
    See

    >>>    1  |import copy
           2  |import math
           3  |import types
           4  |import warnings
```

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -1496,7 +1497,7 @@ def __init__(
             for lr, group in zip(base_lrs, optimizer.param_groups):
                 if isinstance(group["lr"], Tensor):
                     lr_val = lr.item() if isinstance(lr, Tensor) else lr
-                    group["lr"].fill_(lr)
+                    group["lr"].fill_(lr_val)
```
Comment (vadimkantorov) [inline torch/optim/lr_scheduler.py:1500]: btw, wouldn't `fill_` directly accept scalar tensor values? if so, maybe `.item()` not needed? or is the problem with device mixing (`cuda_tensor.fill_(scalar_cpu_tensor)` or `cpu_tensor.fill_(scalar_cuda_tensor)`)?

Comment (GdoongMathew) [reply torch/optim/lr_scheduler.py:1500]: I think its probably because of the device mixing problem. `.item()` could solve the problem if the `lr` tensor is in devices other than `cpu`.

Comment (vadimkantorov) [reply torch/optim/lr_scheduler.py:1500]: Maybe then it can be replaced as `group["lr"].fill_(lr.to(device = group["lr"].device) if isinstance(lr, Tensor) else lr)` making the device-mixing problem more explicit

Comment (GdoongMathew) [reply torch/optim/lr_scheduler.py:1500]: I think before fixing this, the second question mentioned in https://github.com/pytorch/pytorch/issues/126854#issuecomment-2132123607 needs to be settled? As this question also covers other parameters, such as `max_lr` in `OneCycleLR`, and your solution could be generally applied if the answer to that question is yes.

Comment (vadimkantorov) [reply torch/optim/lr_scheduler.py:1500]: It appears that scalar tensor device mixing works fine with `fill_` - probably `fill_` does a copy under the hood on itself:
```python
import torch
cpu_tensor = torch.tensor(2, device = 'cpu')
gpu_tensor = torch.tensor(1, device = 'cuda')
gpu_tensor.fill_(cpu_tensor)
cpu_tensor.fill_(gpu_tensor)
# completes fine
```

so maybe even `group["lr"].fill_(lr)` could be done without any if checks

Comment (GdoongMathew) [reply torch/optim/lr_scheduler.py:1500]: weird, I thought `fill_` would be like other operations, require two tensors located in the same device.
If this is approved, I'd open another PR for this change~

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
Thanks for hopping on this! Agreed that avoiding item calls is ideal, though should be accompanied by a test case verifying that it works as expected.

**Review by janeyx99 — COMMENTED** [review]
@GdoongMathew I was wrong about clone().detach()! It should just be `clone()`--the `detach()` will break differentiability through the optimizer step which is bad. Please modify this before landing.

Also, I updated what I think the longterm fix should be if you're interested in implementing that: https://github.com/pytorch/pytorch/issues/126854#issuecomment-2139963804 (This part does not need to be in this PR)

**Comment by GdoongMathew** [comment]
@janeyx99 `detach` is removed. Thanks.

**Comment by GdoongMathew** [comment]
Hi @janeyx99 
This two fails seems not related with this PR.
